tests/foreman/cli/test_logging.py

```diff
@@ -0,0 +1,189 @@
+:CaseImportance: Medium
+
+:Upstream: No
+"""
+import re
+import logging
+from robottelo import ssh
+from robottelo.test import CLITestCase
+from robottelo.cli.factory import (
+    make_org,
+    make_product_wait,
+    make_repository,
+)
+from fauxfactory import gen_string
+log = logging.getLogger(__name__)
+
+
+class SimpleLoggingTestCase(CLITestCase):
+    """Test class for default logging to files."""
+
+    org = None
+    product = None
+
+    def setUp(self):
+        """Tests for logging to files"""
+        super(SimpleLoggingTestCase, self).setUp()
+
+        # need own org for the manifest refresh test
+
+        if SimpleLoggingTestCase.org is None:
+            SimpleLoggingTestCase.org = make_org(cached=True)
+        if SimpleLoggingTestCase.product is None:
+            SimpleLoggingTestCase.product = make_product_wait(
+                {u'organization-id': SimpleLoggingTestCase.org['id']},
+            )
+
+    def _make_repository(self, options=None):
+        """Makes a new repository and asserts its success"""
+        if options is None:
+            options = {}
+
+        if options.get('product-id') is None:
+            options[u'product-id'] = self.product['id']
+
+        return make_repository(options)
+
+    def test_positive_logging_from_foreman_core(self):
+        """Check that GET command to Hosts API is logged and has request ID."""
+        GET_line_found = 0
+        # log the start of the test in case of problems with slow systems and time zones
+        self.logger.info("Testing test_positive_logging_from_foreman_core")
+        with ssh.get_connection() as connection:
+            result = connection.run('hammer host list')
+            self.assertEqual(result.return_code, 0)
+            # extract last ten lines from log
+            result = connection.run('tail -20 /var/log/foreman/production.log > \
+                        /var/tmp/logfile_from_foreman_core')
+            self.assertEqual(result.return_code, 0)
+        # use same location on remote and local for log file extract
+        logfile_location = '/var/tmp/logfile_from_foreman_core'
+        ssh.download_file(logfile_location)
+        # search the log file extract for the line with GET to host API
+        with open("/var/tmp/logfile_from_foreman_core", "r") as logfile:
+            for line in logfile:
+                if re.search(r'Started GET \"\/api/hosts\?page=1', line):
+                    self.logger.info('Found the line with GET to hosts API')
+                    GET_line_found = 1
+                    # Confirm the request ID was logged in the line with GET
+                    match = re.search(r'\[I\|app\|\w{8}\]', line)
+                    if match:
+                        self.logger.info("Request ID found for logging from foreman core")
+                    else:
+                        assert match == 1, "Request ID not found"
+        assert GET_line_found == 1, "The GET command to list hosts was not found in logs."
```

Comment:
Perhaps it's all edge cases, but some ideas when that assertion will produce incorrect result:

* another test generates a lot of data between us calling "GET" and getting log for further processing
* Satellite throws exception on receiving GET and saves it in log file
* Satellite starts logging much more data for each request and 20 lines are not enough
* Satellite never receives our GET, but some other test asked for list of hosts recently and their request is still in log

These can be mitigated by sending unique request (like asking for host with specific name, one that we created a while earlier) and reading logs from last X minutes, no matter how many lines there are.

Reply:
1] For the more verbose candlepin logs @JacobCallahan suggested the use of sed to cut the lines from the log that were written during the test. I have now implemented that method for the candlepin log test and I could use that method in the other tests if required.

2] The important thing we are looking for is the ID in the first line of an event. This was fixed recently and is essential for following logs using logging software. The ID was present from 2nd line only before, which made it very difficult to follow logs in busy system.

3] I can use the same method as for candlepin test if required.

4] I think we do not care about that. See point 2.

Thank you

Reply:
> 2] The important thing we are looking for is the ID in the first line of an event. This was fixed recently and is essential for following logs using logging software. The ID was present from 2nd line only before, which made it very difficult to follow logs in busy system.

Yes, but you are processing only last 20 lines of `/var/log/foreman/production.log`. And I mean there is a very slight chance these 20 lines will not contain first line of an event. You need something like this sed cutting to ensure you obtain all lines you need, regardless of how many that is.

I don't really require you to implement these changes, this is more food for thought.

Reply:
ACK, now that I have that sed code working I can use everywhere rather than tail. Will come back to this.

Reply:
I have updated all tests to use sed method same as the candlepin test